lib/bap_powerpc/bap_powerpc.mli

```diff
@@ -0,0 +1,614 @@
+
+(**
+
+   ## Intro
+
+    This module is the only one that needed to write lifter functions.
+    The main idea is to make a life of lifter writers as easy as
+    possible. This implies that only a few lines of code should be
+    enough to describe any instruction.
+
+    We introduce RTL - the language we expect to be very expressive,
+    with lots of details hidden under the hood, so a user should not
+    care about minor details.
+
+    So proposed usage is just to open it at the very beginning of your
+    module:
+
+    open Bap_powerpc.Std
+    ...
+
+    Note, that everywhere in this module, where some operation has
+    a notion of bit position (byte/word ...), it's assumed that
+    numeration starts from most significant bit (byte/word ...).
+
+
+   ## Lifter
+
+    Any lifter function must have two arguments.
+    The first one is a model of CPU that contains all information like
+    registers, memory access, instruction address, etc.
+    The second argument is an array of instruction operands, and its
+    user responsibility to treat each element of this array
+    according to instruction semantics. Those, for add instruction
+    which sum register and immediate and save a result in a register,
+    a user must point that the first and the second element of
+    operands array registers and the third argument is immediate.
+
+    In general, what one should do to add an instruction is to write
+    a lifter function and to add it to others.
+
+
+   ## RTL
+
+    The central part of this module is RTL. It contains expressions,
+    operations over expressions and statements.
+    Basically, any line that ended with ';' is a statement,
+    and any part of it is either expression(s), or operation over
+    expression(s).
+
+    ### Bitwidth and Signess
+    Any expression in RTL has a notion of signess. Any operation over
+    expressions of different sign causes a casting to signed.
+
+    Any expression in RTL has a notion of bitwidth. All operations
+    over expressions of different width will cause a casting to the
+    more biggest bitwidth.
+
+    Note, that any expression that tends to take part of other
+    expression is always unsigned. The same is true for any
```

Comment:
I can't understand this question. What do you mean by "an expression that tends to take a part of other expression"?